dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1608,6 +1608,7 @@ bool DeltaMergeStore::handleBackgroundTask(bool heavy)
         }
         case TaskType::Compact:
             task.segment->compactDelta(*task.dm_context);
+            task.segment->placeDeltaIndex(*task.dm_context);
```

Comment:
After `compactDelta`, it may be not a good point to `placeDeltaIndex`.  
Because after flush task complete, it indicates that there may some fresh data at the tail which just flush from memory to disk. So it's a good signal to try to `placeDeltaIndex`. 
But `compactDelta` is just a reorganization of the old data, so I suggest not adding this line here.